src/main/java/com/google/javascript/gents/ModuleConversionPass.java

```diff
@@ -134,16 +149,43 @@ public void visit(NodeTraversal t, Node n, Node parent) {
           // We export the longest valid prefix
           String exportedNamespace = nameUtil.findLongestNamePrefix(lhs, symbols.keySet());
           if (exportedNamespace != null) {
-            convertExportAssignment(child, exportedNamespace, symbols.get(exportedNamespace));
+            convertExportAssignment(
+                child, exportedNamespace, symbols.get(exportedNamespace), filename);
             // Registers symbol for rewriting local uses
-            registerLocalSymbol(child.getSourceFileName(), exportedNamespace, exportedNamespace,
+            registerLocalSymbol(
+                child.getSourceFileName(),
+                exportedNamespace,
+                exportedNamespace,
                 symbols.get(exportedNamespace));
           }
           break;
         default:
           break;
       }
     }
+
+    private void directlyExportSymbol(Node namedNode, Node parent, String filename) {
+      if (!fileToModule.containsKey(filename)) {
+        return;
+      }
+      FileModule module = fileToModule.get(filename);
+      Node child = namedNode.getFirstChild();
+      String nodeName = checkNotNull(child.getQualifiedName());
+
+      ExportedSymbol exportedSymbol = ExportedSymbol.of(filename, nodeName, nodeName);
+      if (module.exportedNamespacesToSymbols.containsValue(nodeName)) {
+        Node next = namedNode.getNext();
+        namedNode.detachFromParent();
+
+        Node export = new Node(Token.EXPR_RESULT, new Node(Token.EXPORT, namedNode));
+        nodeComments.moveComment(namedNode, export);
+        parent.addChildBefore(export, next);
+        directlyExportedSymbols.add(exportedSymbol);
+        compiler.reportCodeChange();
+      } else {
+        exportsToNodes.put(exportedSymbol, namedNode);
```

Comment:
add a comment what condition this represents? I think I'd also pull this up by inverting the if/else cases, and then doing an early exit from the function to reduce indentation and make it easier to understand ("first handle special case X and return, then do the rest...").